### PR TITLE
Dev abhi

### PR DESCRIPTION
**Pull request by ghost**
**PR includes:**
- blog page added
- UI design updated

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
-import Layout from "./components/layout/Layout";
+import React from 'react'
+import Layout from './layout/Layout'
```
Comment (Souvikdeb2612) [inline src/App.js:4]: 1 line space above internal imports

File: src/components/blog/BlogArticle.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import React, { useContext } from "react";
+import { BlogContext } from "./BlogContext";
```
Comment (Souvikdeb2612) [inline src/components/blog/BlogArticle.jsx:4]: same as above

File: src/components/blog/BlogArticle.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    <br />
+                    <span className="blog__country">{blog.country}</span>
+                    <span className="blog__published">{blog.pubDate}</span>
+                  </div>
+                </div>
+              </>
+            );
+          })
+        ) : (
+          <h1 className="loading__text">Loading....</h1>
```
Comment (Souvikdeb2612) [inline src/components/blog/BlogArticle.jsx]: Use a custom loader library for this

File: src/components/blog/BlogArticle.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      <div className="header__text">
+        <h1>Crypto Blog</h1>
+        <p>Get crypto market updates every day</p>
+      </div>
+
+      <div className="separator"></div>
+
+      <div className="all__blogs">
+        {data ? (
+          data.results.map((blog) => {
```
Comment (Souvikdeb2612) [inline src/components/blog/BlogArticle.jsx]: destructure blog here

File: src/components/blog/BlogContext.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  const apiKey = `pub_13393cacdeb562c21d5ce9a7c508c26dc2441`;
+
+  useEffect(() => {
+    axios
+      .get(
+        ` https://newsdata.io/api/1/news?apikey=${apiKey}&q=Cryptocurrency&language=en `
+      )
+      .then((response) => setData(response.data))
+      .catch((error) => console.log(error));
```
Comment (Souvikdeb2612) [inline src/components/blog/BlogContext.jsx]: handle error from UI and not from console

File: src/components/sidebar/Sidebar.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import React, { useState } from "react";
+import { SidebarData } from "../../data/Data";
+import { NavLink } from "react-router-dom";
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx:2]: format imports

File: src/components/sidebar/Sidebar.jsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      left : '-100%'
+    }
+  };
+
+  return (
+    <>
+      <div
+        className="menu-bars"
+        style={expanded ? {left: '30%'} : {left: '2%'}}
+        onClick={() => setExpaned(!expanded)}
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: create a function and call here

File: src/components/sidebar/Sidebar.jsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        <div className="logo">
+          <RiHandCoinLine />
+          <span className="logo-text">
+            Crypto<span>Trade</span>
+          </span>
+        </div>
+
+        <div className="nav-menu">
+          <div className="links">
+            {SidebarData.map((item, index) => (
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: destructure item here

File: src/components/sidebar/Sidebar.jsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+              <NavLink
+                to={item.path}
+                key={index}
+                className={(navClass) =>
```
Comment (Souvikdeb2612) [inline src/components/sidebar/Sidebar.jsx]: you can directly write (isActive)=>isActive ?

File: src/layout/Layout.jsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import Sidebar from "../components/sidebar/Sidebar";
```
Comment (Souvikdeb2612) [inline src/layout/Layout.jsx:3]: 1 line space above internal imports

File: src/pages/Blog.jsx [hunk 11]
```diff
@@ -1,9 +1,15 @@
-import React from 'react'
+import React from "react";
+import BlogArticle from "../components/blog/BlogArticle";
```
Comment (Souvikdeb2612) [inline src/pages/Blog.jsx]: same as above

File: src/routes/Router.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { Routes, Route, Navigate } from 'react-router-dom'
+import Blog from '../pages/Blog'
```
Comment (Souvikdeb2612) [inline src/routes/Router.js:4]: same as above

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/sidebar/Sidebar.jsx (hunk 9 above)
Comment (Pro-Abhi) [reply src/components/sidebar/Sidebar.jsx:55]: **(isActive)=>isActive ?**
using this method getting unexpected output

File: src/layout/Layout.jsx (hunk 10 above)
Comment (Pro-Abhi) [reply src/layout/Layout.jsx:2]: solved

File: src/pages/Blog.jsx (hunk 11 above)
Comment (Pro-Abhi) [reply src/pages/Blog.jsx:2]: solved

File: src/routes/Router.js (hunk 12 above)
Comment (Pro-Abhi) [reply src/routes/Router.js:3]: solved

File: src/components/sidebar/Sidebar.jsx (hunk 6 above)
Comment (Pro-Abhi) [reply src/components/sidebar/Sidebar.jsx:3]: solved

File: src/components/blog/BlogArticle.jsx (hunk 4 above)
Comment (Pro-Abhi) [reply src/components/blog/BlogArticle.jsx:22]: solved

File: src/components/blog/BlogArticle.jsx (hunk 3 above)
Comment (Pro-Abhi) [reply src/components/blog/BlogArticle.jsx:52]: solved

File: src/components/blog/BlogArticle.jsx (hunk 2 above)
Comment (Pro-Abhi) [reply src/components/blog/BlogArticle.jsx:2]: solved

File: src/App.js (hunk 1 above)
Comment (Pro-Abhi) [reply src/App.js:2]: solved

File: src/components/sidebar/Sidebar.jsx (hunk 8 above)
Comment (Pro-Abhi) [reply src/components/sidebar/Sidebar.jsx:51]: destructuring works for an only the title not for the icon

File: src/components/blog/BlogArticle.jsx (hunk 4 above)
Comment (Souvikdeb2612) [reply src/components/blog/BlogArticle.jsx]: destructure it in the same line while mapping

File: src/routes/Router.js (hunk 12 above)
Comment (Souvikdeb2612) [reply src/routes/Router.js:4]: Not solved

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

**Review by Souvikdeb2612 — APPROVED** [review]

**Pull request merged**